Added the possibility of passing custom headers

huguesalary opened the pull request:
Using the `-H` command line switch

-  had to refactor the lamdba.go to use flags instead of os.Args[], to simplify the identification of the arguments being passed and to make it easier to pass multiple times the same flag

huguesalary commented:
I realize I should probably add some code checking that the headers being passed in a valid way (that is, `Header: value`)

gkeiser commented:
Just wanted to vote this one up.  I pulled down the fork and it works very nicely!

jcxplorer commented:
@huguesalary Thanks for this, and for refactoring lambda.go to use flags :)

Having the headers validated would be nice. Do you want to do that as part of this PR or send a separate one? If you send a separate one, I can merge this as-is.

huguesalary commented:
I would actually wait before merging. I realize that besides that fact that I do not validate the headers, other things are wrong with my code.

For example, in `lambda.go` my flag variables are global, which is probably not necessary, nor a good idea.

I'll try to fix those issues soon.

huguesalary commented:
I just modified my PR by adding code checking that the header passed are of the form `Header: value` and changed previously global variables into local variables.

I think the code is fine enough to be merged now.

matiaskorhonen commented:
@jcxplorer you want to review this?

jcxplorer commented on lambda/lambda.go:
```diff
@@ -196,13 +214,21 @@ func runLoadTest(client *http.Client, sqsurl string, url string, totalRequests i
 	for _ = range jobs {
 		start := time.Now()
 		req, err := http.NewRequest(requestMethod, address, bytes.NewBufferString(requestBody))
+		if err != nil {
```
Should not proceed with the request if this fails.

jcxplorer replied:
Not related to this PR, but if something like this fails, we need a way of cancelling the entire load test.

jcxplorer commented:
@huguesalary thanks a lot for this! We'll also need to improve the error handing in general in the lambda function to cancel load tests that have unrecoverable errors.
